rpc/json.go

```diff
@@ -119,6 +119,9 @@ func (c *jsonCodec) ReadRequestHeaders() ([]rpcRequest, bool, RPCError) {
 
 	var incomingMsg json.RawMessage
 	if err := c.d.Decode(&incomingMsg); err != nil {
+		if err == io.EOF {
+			return nil, false, ErrConnectionClosed
```

Comment:
The codec returns an `RPCError` since this method can return multiple type of errors with different RPC error codes.